### DISCUSSION TITLE
Missing AWS Lambda trace data for X-Ray. Ensure Active Tracing is enabled and no subsegments are created outside the function handler

### DISCUSSION THREAD

**Discussion by alakdam07** [serverless/serverless#10794] [Q&A] [answered]
I have a lambda function which has written in NodeTS.I enabled the active tracing I can see the xray trace every time I called the function. This Lambda function also publishes to an SNS. I want to invoke the function locally, I follow this [serverless-local-invoke](https://www.serverless.com/framework/docs/providers/aws/cli-reference/invoke-local) and choose the `--path` options. For this I used this command: `yarn serverless invoke local --function new-order-from-sns --stage test --path 'root/src/functions/picking.json'`, It invoked successfully but in try catch block it throws me the error: 

> Missing AWS Lambda trace data for X-Ray. Ensure Active Tracing is enabled and no subsegments are created outside the function handler.","stackTrace":"Error: Missing AWS Lambda trace data for X-Ray. Ensure Active Tracing is enabled and no subsegments are created outside the function handler.\n    at Object.contextMissingRuntimeError [as contextMissing]

I am not sure how to silent my error.

**This is my lambda function**

    import middy from '@middy/core'
    import { SNSEvent } from 'aws-lambda'
    import * as AWSXRay from 'aws-xray-sdk'
    import { groupBy, prop } from 'ramda'

    import { MASKED_LOG_FIELD_NAMES } from '@/backend/products/src/constants/constants'
    import { middySnsErrorHandler } from '@/backend/shared/middleware/middy-sns-error-handler'
    import { middyCorrelationIds } from '@/backend/shared/middleware/wrapper/middy-correlation-ids'
    import { Log, middyLogMask } from '@/shared/logging/shared-logger'

    import { middyXRayEgressHttp } from '../../../shared/middleware/middy-x-ray-egress-http'
    import { initializeOrderState } from '../data-sources/initialize-order-state'
    import { IS_OFFLINE } from '../settings/settings'
    import { Order } from '../types/order'
    import { isPriceDeviationOverThreshold } from '../utils/orders-utils'

    if (IS_OFFLINE) {
      AWSXRay.setLogger(Logger('silent'))
    }

    // handler to process incoming orders and order changes from picking perspective
    const checkForPriceDeviationsOnCapture = (orders: Order[]): void => {
      try {
        orders.forEach((order) => {
          if (isPriceDeviationOverThreshold(order))
            Log.warn('Order price estimate to captured sum deviation', { order })
        })
      } catch (error) {
        Log.warn('Order price estimate to captured sum error', {}, error)
      }
    }

    const initializeOrders = async (orders: Order[]): Promise<void> => {
      Log.info('Initializing order state on order change', { orders })

      try {
        const ordersByStore = groupBy(prop('storeId'), orders)

        console.log({ ordersByStore })

        await Promise.all(Object.keys(ordersByStore).map((k) => initializeOrderState(ordersByStore[k])))
      } catch (error) {
        Log.error('Failed Initializing order state', error)
      }
    }

    const parseOrdersFromSns = (event: SNSEvent): Order[] => {
      try {
        return event.Records.flatMap((r) => (JSON.parse(r.Sns.Message) as unknown) as Order[])
      } catch (error) {
        Log.error('New order from SNS failed at parsing orders', { event }, error)
        return []
      }
    }

    const handlerFn = async (event: SNSEvent): Promise<void> => {
      const orders = parseOrdersFromSns(event)

      console.log({ orders })

      if (orders.length === 0) return
      checkForPriceDeviationsOnCapture(orders)
      await initializeOrders(orders)
    }

    export const handler = middy(handlerFn)
      .use(middyLogMask({ maskedLogFieldNames: MASKED_LOG_FIELD_NAMES }))
      .use(middyCorrelationIds())
      .use(middyXRayEgressHttp())
      .use(middySnsErrorHandler())

**Comment by pgrzesik** [comment, marked as answer]
Hello - maybe you should not be including the XRay middy middleware in case of running locally? I believe the middleware expects the trace to be present and it fails because local invoke does not inject it into invocation.

**Reply by alakdam07** [reply]
Thank you for the reply. I figured it out. In local environment, I ignored the error if X-ray ID is not exist.

**Reply by pgrzesik** [reply]
Glad to hear that you've resolved your problem 🙌

**Reply by hhaddad-ph** [reply]
Alakdam07, do you mind sharing how you fixed this? I'm having the same challenge
